> I m surprised about the non matching posterior variance, can you fix the optimization (higher lr or more iterations for cvi and let me know if the test still fails?)

Here's the disagreement in variance (the same whether running `update_sites` once or 1,000 times):
```python
>       assert np.allclose(gpr_var, st_var, atol=atol, rtol=rtol)
E       assert False
E        +  where False = <function allclose at 0x7f1aa6240440>(<tf.Tensor: shape=(4, 1), dtype=float64, numpy=\narray([[0.45350882],\n       [0.45350882],\n       [0.45350882],\n       [0.45350882]])>, <tf.Tensor: shape=(4, 1), dtype=float64, numpy=\narray([[0.48707519],\n       [0.41994245],\n       [0.48707519],\n       [0.41994245]])>, atol=1e-06, rtol=1e-06)
```
So GPR predicts 4 identical marginal variances of `0.45350882`, while spatiotemporal CVI predicts 2 unique variances `[[0.48707519], [0.41994245]]`, each appearing twice. Interestingly, the average of these two values is `0.45350882`, exactly GPR's prediction.

Varying the learning rate from 0.01 to 1 and varying the number of `update_sites` calls from 10 to 1_000 still causes just the variance test to fail.

--- a/tests/integration/models/test_spatio_temporal_variational.py
+++ b/tests/integration/models/test_spatio_temporal_variational.py
@@ -72,6 +72,10 @@ def _setup_gpr_model_fixture(st_data):
 
 
 def test_spatiotemporalsparsevariational(st_model_params, gpr_model, st_data):
+    """
+    Test that `SpatioTemporalSparseVariational` trained on data at the inducing points
+    evaluated at the inducing points gives the same ELBO and predicted mean as `GPR`.
+    """
     st_model = SpatioTemporalSparseVariational(**st_model_params)
     assert isinstance(st_model, SpatioTemporalBase)
     for t in (
@@ -107,6 +111,10 @@ def test_spatiotemporalsparsevariational(st_model_params, gpr_model, st_data):
 
 
 def test_spatiotemporalsparsecvi(st_model_params, gpr_model, st_data):
+    """
+    Test that `SpatioTemporalSparseCVI` trained on data at the inducing points
+    evaluated at the inducing points gives the same ELBO and predicted mean as `GPR`.
+    """
     st_model = SpatioTemporalSparseCVI(**st_model_params, learning_rate=1.0)
     assert isinstance(st_model, SpatioTemporalBase)
     true_likelihood = gpr_model.log_marginal_likelihood()

--- a/tests/integration/models/test_spatio_temporal_variational.py
+++ b/tests/integration/models/test_spatio_temporal_variational.py
@@ -83,7 +83,7 @@ def test_spatiotemporalsparsevariational(st_model_params, gpr_model, st_data):
 
     true_likelihood = gpr_model.log_marginal_likelihood()
 
-    opt = tf.optimizers.Adam(learning_rate=1e-2)
+    opt = tf.optimizers.Adam(learning_rate=1e-1)
 
     @tf.function
     def opt_step(data):

--- a/tests/integration/models/test_spatio_temporal_variational.py
+++ b/tests/integration/models/test_spatio_temporal_variational.py
@@ -56,12 +56,6 @@ def _setup_spatiotemporal_params_fixture(st_data):
         kernel_time=Matern12(lengthscale=1, variance=1),
         likelihood=Gaussian(),
     )
-    for t in (
-        params["kernel_space"].trainable_variables
-        + params["kernel_time"].trainable_variables
-        + params["likelihood"].trainable_variables
-    ):
-        t._trainable = False
     return params
 
 
@@ -74,14 +68,19 @@ def _setup_gpr_model_fixture(st_data):
         ]
     )
     model = GPR(data=st_data, kernel=kernel)
-    for t in model.trainable_variables:
-        t._trainable = False
     return model
 
 
 def test_spatiotemporalsparsevariational(st_model_params, gpr_model, st_data):
     st_model = SpatioTemporalSparseVariational(**st_model_params)
     assert isinstance(st_model, SpatioTemporalBase)
+    for t in (
+        st_model.kernel.kernel_space.trainable_variables
+        + st_model.kernel.kernel_time.trainable_variables
+        + st_model._likelihood.trainable_variables
+    ):
+        t._trainable = False
+
     true_likelihood = gpr_model.log_marginal_likelihood()
 
     opt = tf.optimizers.Adam(learning_rate=1e-2)

